staging/src/k8s.io/apiserver/pkg/server/options/admission.go

```diff
@@ -219,6 +221,24 @@ func (a *AdmissionOptions) Validate() []error {
 			registeredPlugins.Difference(intersections).List()))
 	}
 
+	// TODO: is there a better location? we need to ensure that the
+	// admission plugin is not disabled by the admin when the feature
+	// gate is enabled, otherwise we skip the check whether the user
+	// has permission to do unsafe deletion of corrupt object
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.AllowUnsafeMalformedObjectDeletion) {
+		// the plugin and the feature have the same name
+		// the plugin is enabled by default, and is active only
+		// when the feature is enabled
+		name := string(genericfeatures.AllowUnsafeMalformedObjectDeletion)
+		// developer error, the plugin should be registered
+		if !registeredPlugins.Has(name) {
+			errs = append(errs, fmt.Errorf("the plugin %q must be registered, and enabled by default", name))
+		}
+		if disablePlugins.Has(name) {
```

Comment:
What is ensuring it is enabled?    Disabled is just a force disabled list and since this is being set a DefaultOff, what turns it on?

Reply:
the plugin is enabled by default, but it is active only when the feature is enabled. I enabled the plugin by removing it from `DefaultOffAdmissionPlugins` :
```
modified   pkg/controlplane/apiserver/samples/generic/server/admission.go
 
 // DefaultOffAdmissionPlugins get admission plugins off by default for kube-apiserver.
@@ -46,6 +47,7 @@ func DefaultOffAdmissionPlugins() sets.Set[string] {
 		ctbattest.PluginName,                 // ClusterTrustBundleAttest
 		certsubjectrestriction.PluginName,    // CertificateSubjectRestriction
 		validatingadmissionpolicy.PluginName, // ValidatingAdmissionPolicy, only active when feature gate ValidatingAdmissionPolicy is enabled
+		unsafedeletepolicy.PluginName,        // AllowUnsafeMalformedObjectDeletion, only active when feature gate AllowUnsafeMalformedObjectDeletion is enabled
 	)
```

Reply:
no longer relevant, resolving it.